ProcessLib/GroundwaterFlowProcess.h

```diff
@@ -329,6 +400,10 @@ class GroundwaterFlowProcess : public Process
     std::unique_ptr<typename GlobalSetup::VectorType> _rhs;
     std::unique_ptr<typename GlobalSetup::VectorType> _x;
 
+#ifdef USE_PETSC
+    std::unique_ptr<typename GlobalSetup::LinearSolver> _linearSolver;
```

Comment:
Yes, it is possible by adding a string type of argument to  ``GlobalSetup::LinearSolver(...) ``. However, in such way, PETSc ksp solver  has to be created  and deleted for every time when it is called, which, you know, would involves a lot of memory allocation/release operations for the computation of transient problems.